tests/ludwig/datasets/model_configs/train_all_model_configs.py

```diff
@@ -0,0 +1,133 @@
+#! /usr/bin/env python
+#
+# Trains a ludwig model for every dataset which has a default_model_config.
+# You must have valid kaggle credentials in your environment, a few GB of disk space, and good internet bandwidth.
+# Also, for each dataset associated with a Kaggle competition you'll need to sign in to Kaggle and accept the terms of
+# the competition.
+#
+import multiprocessing
+import time
+from dataclasses import dataclass
+from typing import Optional
+
+import pandas as pd
+
+from ludwig import datasets, visualize
+from ludwig.api import LudwigModel
+
+
+@dataclass
+class TrainingResults:
+    """Results of a training run for a dataset."""
+
+    dataset_version: str
+    dataset_name: str
+    has_config: bool
+    output_directory: Optional[str] = None
+    splits: Optional[str] = None
+    metric: Optional[str] = None
+    performance: Optional[float] = None
```

Comment:
These two should probably be two lists or a dict mapping from metric to its value. As an example, a regression dataset like `sarcos` has the following metrics:
- `mean_absolute_error`
- `mean_squared_error`
- `r2`
- `root_mean_squared_error`
- `root_mean_squared_percentage_error`

Reply:
I considered this, but I wanted a simple table of outputs to look at, and putting many metrics in a table makes things harder to read.  This isn't a real benchmark, just a quick-and-simple test script which can be run on a laptop.